software/validation/plot_magnitude_grids.py

```diff
@@ -0,0 +1,180 @@
+                    s=5, edgecolor='',
+                    cmap=plt.cm.gist_ncar)
+        plt.colorbar()
+        if i_fig == 0:
+            plt.xlabel('$m_{input}$')
+            plt.ylabel('$m_{fit}-m_{input}$')
+
+        xmax = mag_data[name]['input'].max()
+        xmin = mag_data[name]['input'].min()
+        ymax = mag_data[name]['mag_resid'].max()
+        ymin = mag_data[name]['mag_resid'].min()
+        plt.text(xmax-0.2*(xmax-xmin), ymax-0.1*(ymax-ymin),
+                 name, fontsize=15)
+
+    plt.tight_layout()
+    plt.savefig(os.path.join(output_dir, 'mag_validation_plot.png'))
+    plt.close()
+
+
+    ###plot color residual grid
+    plt.figsize = (30, 30)
+    for i_fig, name in enumerate(name_list):
+        plt.subplot(3, 2, i_fig+1)
+        sorted_dex = np.argsort(fit_data[name]['ct'])
+        plt.scatter(fit_data[name]['mag_resid'][sorted_dex],
+                    fit_data[name]['color_resid'][sorted_dex],
+                    c=np.log10(mag_data[name]['ct'][sorted_dex]),
+                    s=5, edgecolor='',
+                    cmap=plt.cm.gist_ncar)
+        plt.colorbar()
+        if i_fig == 0:
+            plt.xlabel('$m_{fit} - m_{input}$')
+            plt.ylabel('color residual')
+
+        xmax = fit_data[name]['mag_resid'].max()
+        xmin = fit_data[name]['mag_resid'].min()
+        ymax = fit_data[name]['color_resid'].max()
+        ymin = fit_data[name]['color_resid'].min()
+        plt.text(xmax-0.2*(xmax-xmin), ymax-0.1*(ymax-ymin),
+                 name, fontsize=15)
+
+    plt.tight_layout()
+    plt.savefig(os.path.join(output_dir, 'color_validation_plot.png'))
+    plt.close()
+
+
+    ###plot cumulative distributions
+    plt.figsize = (30, 30)
+    text_dict = {}
+    for i_fig, name in enumerate(name_list):
+        plt.subplot(3, 2, i_fig+1)
+        text = name
+
+        xx, yy = get_cumulative(mag_data[name]['mag_resid'], mag_data[name]['ct'])
+
+        five_dexes = np.where(np.abs(mag_data[name]['mag_resid'])>=5.0)
+        n_five = mag_data[name]['ct'][five_dexes].sum()
+
+        text += '\nno cut: %.1e tot, %.1e bad' % (mag_data[name]['ct'].sum(), n_five)
+
+        xx = np.log10(xx+0.001)
+        yy = np.log10(yy)
+
+        baseline = plt.plot(xx, yy, color='k')
+
+        xmax = xx.max()
+        xmin = xx.min()
+        ymax = yy.max()
+        ymin = yy.min()
+
+        cut_labels = {}
+
+        for cut, color in zip(color_cuts, color_colors):
+            xx, yy = get_cumulative(mag_resid_dist[name][cut]['mag_resid'],
+                                    mag_resid_dist[name][cut]['ct'])
+
+            five_dexes = np.where(np.abs(mag_resid_dist[name][cut]['mag_resid'])>=5.0)
+            n_five = mag_resid_dist[name][cut]['ct'][five_dexes].sum()
+
+            text += '\n$\sigma_c<=$ %.2f: %.1e tot, %.1e bad' % \
+            (cut, mag_resid_dist[name][cut]['ct'].sum(), n_five)
+
+            xx = np.log10(xx+0.001)
```

Comment:
This looks like it's to avoid plotting problems when xx = 0, but a comment would help